libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2082,24 +2082,28 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 				".";
 
 			if (isStructConstructorCall)
```

Comment:
There's an assertion up the top of the function which states isVariadic is only valid on `unctionCallKind::FunctionCall`